client/theia-glsp-server/src/node/glsp-server-launcher.ts

```diff
@@ -52,7 +54,11 @@ export class GLSPServerLauncher implements BackendApplicationContribution {
 
     protected startServer(): boolean {
         if (this.launchOptions.jarPath) {
-            let args = ['-jar', this.launchOptions.jarPath, '--port', `${this.launchOptions.serverPort}`];
+            let args = ['-jar', this.launchOptions.jarPath]
```

Comment:
why this change?

Reply:
In that case we do not have to definitions of the port for the backend (see method getPort at client/theia-ecore/src/node/ecore-glsp-server-contribution.ts)

Reply:
I don't get it. Aren't you configuring the port in the launchoptions in `CoffeeGlspLaunchOptions` (which should be renamed btw)?
In the previous code iteration you simply accessed the port from the launchoptions and the same is done in https://github.com/eclipsesource/coffee-editor/blob/73db58f116cd75967a663d0a758545ef484031f9/web/coffee-server/src/node/glsp-server-launcher.ts#L55
I think the only necessary change to the previous commit was the change in EcoreServerLauncher.java

Reply:
That is true but in that case I configure in the client/ the ecore-glsp port twice:
Once in the `CoffeeGlspLaunchOptions` .

But the port is also specified as the command line argument that starts theia. (See browser-app/package.json) This is read in the  `ecore-glsp-server-contribution.ts`. 

In my opinion it makes only sense to define this at one place.

Reply:
@tortmayr do we need the parameter in the package.json?
I would prefer to only keep the value in the launchoptions and get rid of the package.json .

Reply:
No we do not need this parameter.  Keeping the value from the launch options is enough.

I had a quick look at the code and have some additional comments:
- The `theia-glsp-server` module can be merged into the `theia-ecore` module. Splitting this has no real benefit. 
- The jarfiles should not be part of the "src" directory and should be located in a separate directory. (e.g. In the coffee-editor the jars are located a "build" directory)
- In general I think it's not a good idea to check the binary jarfiles in. It would be better to add them to .gitignore and simply always copy the generated jarfile during the build.

Reply:
How do I configure the Ecore-Glsp-Client on which port it should connect, if I remove the `--ECORE_GLSP` from the command line? 

I've understood that the port is read from the Command Args during the creation of the Websocket from Clientside at https://github.com/eclipsesource/ecore-glsp/blob/8e1b95ee69d3de3ed96c7b2fca32f6ae811827f9/client/theia-ecore/src/node/ecore-glsp-server-contribution.ts#L25

Reply:
do it just as was done before by reading the config, see https://github.com/eclipsesource/coffee-editor/blob/73db58f116cd75967a663d0a758545ef484031f9/web/coffee-server/src/node/glsp-server-launcher.ts#L55

Reply:
I'm talking about the `BaseGLSPServerContribution` not the `GLSPServerLauncher`.
What's irritating me is that we have not had a LaunchOptions before.
Maybe it would be interesting to move `GLSPLaunchOptions` into GLSP at some point.
Anyways, I will inject this in both mentioned classes and read the port from there.

Reply:
Please have a look on the code again, if this is now in a state to merge, I will squash the branch one more time :)

Reply:
@jfaltermeier could you check this?